### ISSUE TITLE
Undefined property error

### ISSUE THREAD

**Issue by metuza** [facebookincubator/facebook-for-woocommerce#488]
With latest version released today i get undefined property error in line 441 of file "facebook-for-woocommerce/includes/fbproduct.php". And all my products lost their brand name which was replaced by site name.

```
I cant see that use of $this->ID is correct, shouldnt it be $id ?
$brand = get_the_term_list($this->ID, 'product_brand', '', ', ');

I am using the below which works very good:

$brand_terms = wp_get_post_terms( $id, 'product_brand' );
    foreach ( $brand_terms as $brand_term ) $brands[] = $brand_term->name;

    $product_data = array(
      'name' => WC_Facebookcommerce_Utils::clean_string(
        $this->get_title()),
      'description' => $this->get_fb_description(),
      'image_url' => $image_urls[0], // The array can't be empty.
      'additional_image_urls' => array_filter($image_urls),
      'url'=> $product_url,
      'category' => $categories['categories'],
      'brand' => isset( $brand_term->name ) ? $brand_term->name : WC_Facebookcommerce_Utils::get_store_name(),
```

**Comment by sammul** [comment]
[Note : This is an Automated Message]

Hello,

Thanks for your interest in our plugin!
Due to a high volume of issues we may not be able to get to yours immediately.

If this is a feature request, or feedback, please reply with
`Feature Request` or `Feedback` and it will be categorized accordingly, and
we will read it and reply when we can.

If you are experiencing problems, read on.

First. Please check to see if your issue is not a duplicate of another one.
To search our issues, use the search feature at the top of the GitHub issues page.
We have nearly 500 existing and past issues, so it is fairly likely someone else has already reported your problem.
If you do find a relevant issue, please comment in that issue, and close this one.
It helps us prioritize issues to tackle by looking at the number of people with the same problem.

If your issue is unique, and you are sure, then read on.
We require the following information in order to help you:
- WooCommerce Version
- Wordpress Version
- Plugin Version. Please Note that we do not support outdated versions of the plugin.
- A Screenshot of where you see the problem. If it is on Facebook, a screenshot of Facebook,
if it is on WooCommerce, a screenshot of WooCommerce. If it is in multiple places, multiple screenshots.
- The URL of your website.
- The URL of your page.
- The URL(s) where you took your screenshot(s).
- (If applicable) Detailed steps you took in order to see the problem.

Please be advised, it may take us some time to get to your issue.
In the meantime, below are generic approaches you can take that may solve the issue :
- Upgrade to the latest version.
- Force Resync
- Reset the plugin using `Reconfigure > Advanced Options > Delete` and setup again.
- Try to deactivate other plugins which you think could be interfering and repeating the above.

When you have the above data for us, please include
`Debug Ready` in your reply. We will get to it and debug it as soon as we can.

Thanks for your patience,
Dev Team (@dmitridr, @mengyingdu, @sammul, @jieminz22)

**Comment by sammul** [comment]
@metuza Thanks for sharing the code and testing that. You are welcomed to submit the code for review if you are interested. Our team will look into this and fix in next version when confirmed.

**Comment by mengyingdu** [comment]
This will be added in the next release. Thanks @metuza !

**Comment by dmitridr** [comment]
Fixed in https://github.com/facebookincubator/facebook-for-woocommerce/commit/12e43ad11e9effe061306a0f82784f97b1d4e36c as Mengying said, should be in 1.8.8

**Comment by mengyingdu** [comment]
Fixed in new release [v1.9.0](https://github.com/facebookincubator/facebook-for-woocommerce/releases/tag/v1.9.0)
